camayoc/tests/qcs/api/v1/scanjobs/test_run_scanjobs.py

```diff
@@ -303,3 +324,56 @@ def test_scan_task_results(scan_info):
         num_failed = task['systems_failed']
         num_scanned = task['systems_scanned']
         assert num_scanned == sys_count - num_failed
+
+
+@pytest.mark.parametrize(
+    'scan_info', scan_info(), ids=utils.name_getter)
+def test_products_found_deployment_report(scan_info):
+    """Test that the detail report contains expected data for the source.
+
+    :id: d5d424bb-8183-4b60-b21a-1b4ed1d879c0
+    :description: Test that the detail report contains expected data.
+    :steps:
+        1) Retreive the json report for the scan.
+        2) Assert that any products marked as present are expected to be found
+           as is listed in the configuration file for the source.
+    :expectedresults: There are inspection results for each source we scanned
+        and any products found are correctly identified.
+    """
+    result = get_scan_result(scan_info['name'])
+    report_id = result['report_id']
+    report = api.Client().get(
+        'reports/{}/deployments'.format(report_id)).json().get('report')
+    for entity in report:
+        present_products = []
+        all_found_products = []
+        for product in entity.get('products'):
+            name = ''.join(product['name'].lower().split())
+            if product['presence'] == 'present':
+                present_products.append(name)
+            if product['presence'] in ['present', 'potential']:
+                all_found_products.append(name)
+        for source_to_product_map in result['expected_products']:
+            src_id = list(source_to_product_map.keys())[0]
+            hostname = result['source_id_to_hostname'][src_id]
+            ex_products = source_to_product_map[src_id]
+            expected_product_names = [
+                ''.join(name.lower().split('-')) for name in ex_products.keys()
```

Comment:
I see a many lines dealing with converting Camayoc's config file product name to quipucords report product name, I think we could do better here and have a mapping or try to use the same naming convention. What do you think?

I am saying that because we should have somewhere how to properly map the values to help other trying to implement similar tests and follow the same convention.

Reply:
Yeah...you think the yaml would play nice with "JBoss EAP" as a key?

Also could just use lowercase no spaces versions of what they use in quipucords so everything ends up lowercase and with no spaces

Reply:
Using all lowercase and no space seems good to me:  `quipucords_name.lower().replace(' ', '')`

Reply:
I will update that in here and on the jenkins config and alert all our compatriots of the change before merge.